android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java

```diff
@@ -362,70 +362,97 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
                 UserPrefs.get(Profile.getLastUsedRegularProfile())
                         .setBoolean(Pref.HTTPS_ONLY_MODE_ENABLED, newValueBool);
             }
+        } else if (PREF_HTTPS_FIRST_MODE.equals(key)) {
+            UserPrefs.get(Profile.getLastUsedRegularProfile())
+                    .setBoolean(Pref.HTTPS_ONLY_MODE_ENABLED, (boolean) newValue);
         } else if (PREF_DE_AMP.equals(key)) {
             UserPrefs.get(Profile.getLastUsedRegularProfile())
                     .setBoolean(BravePref.DE_AMP_PREF_ENABLED, (boolean) newValue);
         } else if (PREF_DEBOUNCE.equals(key)) {
             UserPrefs.get(Profile.getLastUsedRegularProfile())
                     .setBoolean(BravePref.DEBOUNCE_ENABLED, (boolean) newValue);
         } else if (PREF_IPFS_GATEWAY.equals(key)) {
-            BravePrefServiceBridge.getInstance().setIpfsGatewayEnabled((boolean) newValue);
+            if (BravePrivacySettingsIPFSUtils.isIPFSSupported()) {
```

Comment:
I would suggest to have  at `onPreferenceChange` some trivial methods, like single line
```
someClass.setProperty((boolean) newValue);
```

It was before with `BravePrefServiceBridge`, which we are trying to get rid of, so the idea to change it is good.

If we will keep as it is now here with `boolean => IPFSResolveMethodTypes enum => UserPrefs.setInteger` - then when we'll need in some other place, like infobar or popup message, most probably all this logic wil be copy-pasted.

I would recommend to move this logic into `android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettingsIPFSUtils.java` , and don't remove the class `BravePrivacySettingsIPFSUtils` it as I mentioned in previous comment.

Reply:
Do you mean where we get the value of the preference? I searched Java code and I din't find anywhere this is read, it is only written by the settings. Or do you mean something else?

Reply:
Actually nvm, I can just make two util methods.

Reply:
yes,  currently it is used only here, but in the future this code  for sure may be pasted to other places.

Reply:
Updated.